### PR TITLE
Radio max range, build a camp radio tower

### LINKED ISSUE #28393: Camp Communications Tower
**Is your feature request related to a problem? Please describe.**  
So, when I was scavenging through the city, I happened upon some radios and subsequently thought that it would be pretty neat if you could call one of your camps and arrange missions away from the base. I like the option to build little supply points, but it seems like there is no way to do so without first going to the camp, talking to the manager, finding things to pass the time while things are set up, then returning to the manager to create the site or relay the objects. This is a tedious process. 

**Describe the solution you'd like**  
This is where the criminally underused two way radio comes in. Spread all throughout the towns and cities you will find these books called "Ham Radio Illustrated" and "Amateur Home Radio for Enthusiasts", and I doubt that the pc is the only person to pick up a copy and use what is inside. I would think that there would be many more people and even factions would be communicating via the radio throughout the area. 

This brings me back the two way radio. You could use it as a method to contact your base manager at whichever camp you have built a certain building, or finished a center base upgrade on. Maybe, that same building could be used to do various other actions that a radio station would perform, such as signalling for survivors to come to your camp, (which could attract roving raiders) or contacting the refugee center or other survivor groups to trade or get missions.

**Describe alternatives you've considered**  
Currently, I cannot think of any, but I will edit this post if I do think of any.

**Additional context**  
Currently, I cannot think of any, but I will edit this post if I do think of any.

### PR DESCRIPTION
**Pull request by ghost**
#### Summary

```SUMMARY: Features "Adds radio max range, and radio tower construction"```

#### Purpose of change
Fixes #28393

#### Describe the solution
This somewhat relies on #29249 and has the same code.
Adds a max radio range to the two-way radio ( 200 omts by square_distance() )
This can be extended the higher z-elevation you are. ( or the recieving NPC is - or both )
can build a radio tower as a faction camp upgrade, that boosts the max_range of contacting that NPC if they are at that camp.
Also to fix a previous oversight ive made radio contact impossible whilst deaf or underwater.

#### Describe alternatives you've considered
A proper json-driven implementation of radios and radio ranges and distance calculations would be preferable, that can be incrementally improved.

#### Additional context
N/A

### CODE REVIEW

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by davidpwbrown** [comment]
Needs some refinement, most camp missions will behave strangely when not loaded in reality bubble. needs camp structure to accomodate that first. I Was a bit naieve in this implementation

File: data/json/recipes/recipe_construction_om.json [hunk 1]
```diff
@@ -306,6 +306,64 @@
...
+      { "id": "WRENCH", "level": 1 },
+      { "id": "SCREW", "level": 1 }
+    ],
+    "components": [
+      [
+        [ "wind_turbine", 4 ],
+        [ "xl_wind_turbine", 1 ],
+        [ "solar_panel", 4 ],
+        [ "reinforced_solar_panel", 4 ],
+        [ "reinforced_solar_panel", 4 ],
```
Comment (Night-Pryanik) [inline data/json/recipes/recipe_construction_om.json]: Duplicate?

**Comment by davidpwbrown** [comment]
merge problems, will add a new PR

**Pull request closed without merging**